src/Raven.Embedded/RavenEmbedded.cs

```diff
@@ -0,0 +1,207 @@
+        {
+            // Start the server
+
+            if (Interlocked.CompareExchange(ref _initialized, 1, 0) != 0)
+                throw new InvalidOperationException("Already have been initialized");
+
+            options = options ?? ServerOptions.Default;
+            _serverTask = RunServer(options);
+        }
+
+        public static RavenEmbedded Instance = new RavenEmbedded();
+
+        public async Task<IDocumentStore> GetDocumentStore(string database)
+        {
+            if (Interlocked.Read(ref _initialized) == 0)
+                throw new InvalidOperationException("Please run StartServer before trying to get document store");
+
+            if (string.IsNullOrEmpty(database))
+                throw new ArgumentException(nameof(database));
+
+            if (Logger.IsInfoEnabled)
+                Logger.Info($"GetDocumentStore for db ${ database }.");
+
+            var serverUrl = await GetServerUri();
+
+            if (_documentStores.TryGetValue(database, out var existingStore))
+            {
+                return existingStore;
+            }
+
+            var store = new DocumentStore
+            {
+                Urls = new[] { serverUrl.AbsoluteUri },
+                Database = database
+
+            }.Initialize();
+
+            try
+            {
+                await store.Maintenance.Server.SendAsync(new CreateDatabaseOperation(new DatabaseRecord(database))).ConfigureAwait(false);
+            }
+            catch (ConcurrencyException)
+            {
+                // Expected behaviour when the database is already exists
+
+                if (Logger.IsInfoEnabled)
+                    Logger.Info($"{database} already exist.");
+            }
+
+            _documentStores.TryAdd(database, store);
+            return store;
+
+        }
+
+        public async Task<Uri> GetServerUri()
+        {
+            if (Interlocked.Read(ref _initialized) == 0)
+                throw new InvalidOperationException("Please run Setup before trying to get the server uri");
+
+            return await _serverTask.ConfigureAwait(false);
+        }
+
+        public void RemoveStore(string database)
+        {
+            if(_documentStores.TryRemove(database, out var store))
+                store.Dispose();
+        }
+
+        public void KillSlavedServerProcess()
+        {
+            var p = _ravenDbSlavedProcess;
+            _ravenDbSlavedProcess = null;
+
+            if (p == null || p.HasExited != false)
+                return;
+
+            if (Logger.IsInfoEnabled)
+                Logger.Info($"Kill global server PID { p.Id }.");
+
+            try
+            {
+                p.Kill();
+                Interlocked.CompareExchange(ref _initialized, 0, 1);
+            }
+            catch (Exception e)
+            {
+                if (Logger.IsInfoEnabled)
+                {
+                    Logger.Info($"{DateTime.Now}: {e}\r\n");
```

Comment:
Pass exception normally, and there is no need for the formatting/ exception there.